linkis-public-enhancements/linkis-script-dev/linkis-storage-script-dev-server/src/main/java/org/apache/linkis/filesystem/restful/api/FsRestfulApi.java

```diff
@@ -518,23 +518,35 @@ public Message fileInfo(
       throw WorkspaceExceptionManager.createException(80012);
     }
     FileSource fileSource = null;
+    FileSource fileSource2 = null;
     try {
       Message message = Message.ok();
       fileSource = FileSource$.MODULE$.create(fsPath, fileSystem);
       Pair<Object, Object>[] fileInfo = fileSource.getFileInfo(pageSize);
       IOUtils.closeQuietly(fileSource);
       if (null != fileInfo && fileInfo.length > 0) {
+        int rowNumber = (int) fileInfo[0].getSecond();
         message.data("path", path);
         message.data("colNumber", fileInfo[0].getFirst());
         message.data("rowNumber", fileInfo[0].getSecond());
+        if (rowNumber >= pageSize) {
+          fileSource2 = FileSource$.MODULE$.create(fsPath, fileSystem);
+          fileSource2.collect();
+          message.data("totalLine", fileSource2.getTotalLine());
+          IOUtils.closeQuietly(fileSource2);
```

Comment:
do in finally block

Reply:
optimize filesource2 
add tool function of getTotleLine at org.apache.linkis.storage.utils.FileSystemUtils

Reply:
done